### PR TITLE
linked endpoint for rate and review Podcasts

### LINKED ISSUE #52: As a user, I want to rate podcasts with star ratings and reviews to assist others in discovering quality content.
**Acceptance Criteria:**  

1.  Overall podcast rating is an accurate average of user ratings.
2.  Users can submit text reviews alongside their ratings.
3. All ratings and reviews for the podcast are displayed on its info page.
4. Mechanisms are in place to detect and handle fraudulent or abusive ratings and reviews. Users can report inappropriate content.

**Demo Steps:**  
1. During a podcast episode playback, users are presented with a rating widget.
5. Users can assign a rating to the podcast using a bird rating system similar to 1-5 star system (happy, smiling, mutual, dissatisfied, sad Awaazo bird ).
6. An option is available for users to write a review.
7. Upon submission, ratings and reviews are publicly visible.

**Potential Risks:**  
1. Inconsistent access to the rating and review feature across various devices or browsers.
2. Challenges in ensuring the average rating accuracy due to potential data inconsistencies.
3. The presence of fraudulent reviews may threaten the platform's trustworthiness.

### PR DESCRIPTION
**Pull request by Ahed752**
Enhanced User Interaction with Podcasts: Implemented new features for user engagement, including the ability to add ratings and reviews to podcasts, significantly improving the interactive aspect of the platform.

Refined Podcast Management and Accessibility: Merged various branches to streamline podcast management.

Bug Fixes and Code Refactoring: Addressed critical bugs, particularly in podcast rating functionality.

closes #52

### CODE REVIEW

File: frontend/pages/Main.tsx [hunk 1]
```diff
@@ -5,8 +5,14 @@ import ForYouSection from "../components/home/ForYouSection";
 import ContinueListeningSection from "../components/home/ContinueListening";
 import ExploreGenresSection from "../components/home/ExploreGenres";
 import { podcasts, episodes } from "../utilities/SampleData";
+ 
+
+
 
 const Main = () => {
+
+   // Replace 'test-podcast-id' with an actual ID from your backend
```
Comment (mathieutrudeau) [inline frontend/pages/Main.tsx:14]: That should be removed

**Review by mathieutrudeau — APPROVED** [review]

File: frontend/pages/Main.tsx (hunk 1 above)
Comment (Ahed752) [reply frontend/pages/Main.tsx]: Done

File: Backend/Backend/Controllers/Responses/PodcastResponses.cs [hunk 2]
```diff
@@ -45,11 +45,13 @@ public PodcastResponse(Podcast p, string domainUrl)
...
         Tags = p.Tags;
         IsExplicit = p.IsExplicit;
-        AverageRating = p.AverageRating;
-        TotalRatings = p.TotalRatings;
         Type = p.GetPodcastTypeString();
         PodcasterId = p.PodcasterId;
         Episodes = p.Episodes.Select(e => new EpisodeResponse(e, domainUrl)).ToList();
+        Ratings = p.Ratings.Select(r => new RatingResponse(r)).ToList();
+        TotalRatings = (ulong) Ratings.Where(r=>r.Rating!=0).Count();
+        if(TotalRatings>0)
```
Comment (BigSaladMan) [inline Backend/Backend/Controllers/Responses/PodcastResponses.cs:53]: is this initialised when total ratings are 0? in that case they should be 0

File: Backend/Backend/Models/Podcast.cs [hunk 3]
```diff
@@ -1,56 +1,60 @@
...
-        Tags = Array.Empty<string>();
-        IsExplicit = false;
-        Type = PodcastType.Real;
-        Episodes = new List<Episode>();
-        AverageRating = 0;
-        TotalRatings = 0;       
-    }
-
+    /// <summary>
+    /// Podcast ID
```
Comment (BigSaladMan) [inline Backend/Backend/Models/Podcast.cs:8]: I think you mightve deleted the constructor for this

**Review by BigSaladMan — APPROVED** [review]

File: Backend/Backend/Controllers/Responses/PodcastResponses.cs (hunk 2 above)
Comment (mathieutrudeau) [reply Backend/Backend/Controllers/Responses/PodcastResponses.cs:53]: If total ratings are 0, then an empty list is returned.

File: Backend/Backend/Models/Podcast.cs (hunk 3 above)
Comment (mathieutrudeau) [reply Backend/Backend/Models/Podcast.cs:8]: I did, it was redundant because the properties are already being assigned a default value.

**Pull request merged**